presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushPartialAggregationThroughExchange.java

```diff
@@ -255,4 +266,9 @@ private PlanNode split(AggregationNode node, Context context)
                 node.getHashVariable(),
                 node.getGroupIdVariable());
     }
+
+    private static boolean isLambda(RowExpression rowExpression)
+    {
+        return castToExpression(rowExpression) instanceof LambdaExpression;
```

Comment:
Curious: Why need to cast to expression? We can directly see whether it's a `LambdaDefinitionExpression` right?

Reply:
The problem is we haven't moved the translation up to this rule yet. So, it is actually a `OriginalExpression` which contains a `Expression`.

Reply:
I see, thanks!